proposals/0041-picture-component.md

```diff
@@ -0,0 +1,283 @@
+
+---
+
+### `srcset` support
+
+**User code**
+
+```astro
+---
+import { Image } from "astro:assets";
+import myImage from "../something.png";
+---
+
+<!-- A basic `densities` for 2x, 3x etc is supported -->
+<Image src={myImage} densities={[2, 3]} alt="My image available in 2x and 3x densities" />
+
+<!-- Alternatively, `widths` can be used for precise widths -->
+<Image src={myImage} widths={[1280, 1920]} alt="My image with precise widths" />
+```
+
+**Result**
+
+```html
+<!-- Filepaths omitted for brevity, '...' here would be the usual URLs. -->
+<img src="..." srcset="... 2x, ... 3x" />
+<img src="..." srcset="... 1280w, ... 1920w" />
+```
+
+### Picture component
+
+**User Code**
+
+```astro
+---
+import { Picture } from "astro:assets";
+import myImage from "../something.png";
+---
+
+<!-- Similar to Image, Picture can use either `widths` or `densities`. For brevity, only the later is shown here. -->
+<Picture src={myImage} formats={['avif', 'webp']} densities={[2, 3]} alt="My image available in 3 formats and 3 densities" />
+```
+
+**Result**
+
+```html
+<picture>
+  <source srcset="..., ... 2x, ... 3x" type="image/avif" />
+  <source srcset="..., ... 2x, ... 3x" type="image/webp" />
+  <img
+    src="..."
+    srcset="... 2x, ... 3x"
+    alt="My image available in 3 formats and 3 densities"
+    width="700"
+    height="525"
+    loading="lazy"
+    decoding="async"
+  />
+</picture>
+```
+
+**NOTE:** `Picture` can take all the arguments of `Image`.
+
+**NOTE:** This code represent only the values accepted by the base image services in Astro. A custom image service is free to use other attributes and completely different methods to generate a `srcset` value.
+
+# Background & Motivation
+
+The statement from the original RFC for `astro:assets` still stands: Images on the web are hard! Getting your image, at the perfect quality, in the perfect format, at the perfect size at the specific size you need is, a lot of work!
+
+Abstracting all that work is also a fairly consequential challenge API-design wise, as there's a lot of features to support and concessions to be made for the brevity of the public API while still keeping the underlying API flexible enough.
+
+The main motivation here is again, the same as `astro:assets` itself: Creating a image in multiple formats and multiple sizes in Astro should be as easy as possible, and the API flexible enough for user land to be able to create the missing pieces that the lean core experience cannot fulfil.
+
+# Goals
+
+- Add a hook to generate a `srcset` to image services
+- Add support for generating a `srcset` value for `<Image />`
+- Add a built-in `<Picture />` component, allowing to use multiple formats and image sizes, powered by the image service's ability to generate a `srcset`
+
+# Non-Goals
+
+## For the built-in experience
+
+- Automatic generation or API to make writing `sizes` easier
```

Comment:
You mean we don't append `media`, right? What would someone do who needed this?

Reply:
No, this line is about `sizes` for both `<Image />`, `<Picture />`. Certain frameworks have automatic generation for this, or actually uses `sizes` to generate the `srcset` instead of a separate attribute (typically through a DSL or through a normal `sizes` attribute, convoluted syntax and all)

`media` is also not supported in the base Picture component, because art direction is not supported.

Reply:
`media` and art direction might be a good one to list as alternatives considered

As much as I've reached for that approach over the years, it's pretty cumbersome to work with and is way too easy to silently break your `media` breakpoints later.  Marking that as not supported here sounds like a smart move to me